### PR TITLE
Add typings for ts3-nodejs-library

### PR DESCRIPTION
**Pull request by P4sca1**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an NPM package, match the name. If not, do not conflict with the name of an NPM package.
- [x] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] `tslint.json` should be present, and `tsconfig.json` should have `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

https://github.com/Multivit4min/TS3-NodeJS-Library/issues/46

There are still a lot of `any` types, which should be implemented in the future.

### CODE REVIEW

**Comment by P4sca1** [comment]
Can somebody help me with that?
Whenever I try to run `npm run test` locally it says
```
Error: In package.json: Don't use a 'package.json' for @types dependencies.
    at checkPackageJsonDependencies (/Users/pascal/code/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:199:19)
    at combineDataForAllTypesVersions (/Users/pascal/code/DefinitelyTyped/node_modules/types-publisher/src/lib/definition-parser.ts:99:37)
Error: Parsing failed.
    at fail (/Users/pascal/code/DefinitelyTyped/node_modules/types-publisher/src/util/util.ts:338:20)
    at ChildProcess.child.on (/Users/pascal/code/DefinitelyTyped/node_modules/types-publisher/src/util/util.ts:326:21)
    at ChildProcess.emit (events.js:182:13)
    at finish (internal/child_process.js:797:14)
    at process._tickCallback (internal/process/next_tick.js:61:11)
```.

This stops me from testing locally and I don't want to further spam the CI pipeline.

**Comment by P4sca1** [comment]
I needed to remove `node_modules` and `package-lock.json` and reinstall dependencies from scratch.
The test now runs fine locally.

File: types/ts3-nodejs-library/property/Channel.d.ts [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    /**
+     * Gets the Icon Name of the Channel
+     */
+    getIconName(): Promise<any>;
+
+    /**
+     * retrieves the namespace of this class
+     * @returns the current namespace
+     */
+    getNameSpace(): string;
```
Comment (armanio123) [inline types/ts3-nodejs-library/property/Channel.d.ts]: All methods below `getNameSpace` belongs to `Abstract` with this class already extends so there's no need to implement again. All of the classes extending `Abstract` have this error.

**Review by armanio123 — CHANGES_REQUESTED** [review]

File: types/ts3-nodejs-library/property/Channel.d.ts (hunk 1 above)
Comment (P4sca1) [reply types/ts3-nodejs-library/property/Channel.d.ts]: Resolved via #34797

**Review by armanio123 — APPROVED** [review]

**Comment by P4sca1** [comment]
Thank you for merging this PR!
Could you give me a hint why @types/steam is listed as a dependency on npm? @armanio123
https://www.npmjs.com/package/@types/ts3-nodejs-library

File: types/ts3-nodejs-library/TeamSpeak3.d.ts [hunk 2]
```diff
@@ -0,0 +1,990 @@
+import { EventEmitter } from 'events';
+import TeamSpeakChannel = require('./property/Channel');
+import TeamSpeakClient = require('./property/Client');
+import TeamSpeakServer = require('./property/Server');
+import TeamSpeakServerGroup = require('./property/ServerGroup');
+import TeamSpeakChannelGroup = require('./property/ChannelGroup');
+
+// For the buffer type.
+import * as stream from 'steam';
```
Comment (armanio123) [inline types/ts3-nodejs-library/TeamSpeak3.d.ts]: This might the what's causing your dependency. I think you meant `stream`

**Comment by P4sca1** [comment]
Oh yeah that’s a typo. I will fix it in a new pull request after I added more typings.

File: types/ts3-nodejs-library/TeamSpeak3.d.ts (hunk 2 above)
Comment (P4sca1) [reply types/ts3-nodejs-library/TeamSpeak3.d.ts]: Oh yeah that’s a typo. I will fix it in a new pull request after I added more typings.

**Pull request merged**